src/AutoMapper/QueryableExtensions/ExpressionBuilder.cs

```diff
@@ -18,7 +18,7 @@ namespace AutoMapper.QueryableExtensions
 
     public interface IExpressionBuilder
```

Comment:
I don't consider this a part of our API. I think we should hide `MapperConfiguration.ExpressionBuilder` and only expose it through an implementation interface. I'm sure other members can get the same treatment.

Reply:
@jbogard If you like the idea, perhaps it would be easier to write the code yourself after merging this PR. I'm not sure you would agree with my choices :)

Reply:
Also, we could have an extension method `Advanced()` or smth, in the `AM.Internal` namespace.